### PR TITLE
Store Arrays of temporal values.

### PR DESCRIPTION
**Pull request by sherfert**
This is still missing the storage of DateTimes with ZoneIds (and Arrays thereof), which awaits https://github.com/neo4j/neo4j/pull/10954 to be mergeable.

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/kernel/impl/store/DynamicArrayStore.java [hunk 1]
```diff
@@ -106,7 +113,7 @@ public DynamicArrayStore(
     {
         super( fileName, configuration, idType, idGeneratorFactory, pageCache,
                 logProvider, TYPE_DESCRIPTOR, dataSizeFromConfiguration, recordFormats.dynamic(), recordFormats.storeVersion(), openOptions );
-        allowStorePoints = recordFormats.hasCapability( Capability.POINT_PROPERTIES );
+        allowStorePointsAndTemporal = recordFormats.hasCapability( Capability.POINT_PROPERTIES );
```
Comment (tinwelint) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/store/DynamicArrayStore.java]: did you forget to also add `recordFormats.hasCapability( Capability.TEMPORAL_PROPERTIES )`?

File: community/kernel/src/main/java/org/neo4j/kernel/impl/store/PropertyType.java [hunk 2]
```diff
@@ -159,6 +159,10 @@ else if ( itemType <= GEOMETRY.byteValue() )
             {
                 return headOf( recordBytes, DynamicArrayStore.GEOMETRY_HEADER_SIZE );
             }
+            else if ( itemType <= TEMPORAL.byteValue() )
```
Comment (tinwelint) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/store/PropertyType.java]: The `<=` comparison looks weird. I see that this is most likely a copy-paste issue stemming from the `DOUBLE` comparison above, which `GEOMETRY` also copy-pasted. The `DOUBLE` comparison is, I believe, a special case for covering all integral and floating point types in one comparison, knowing that the `DOUBLE` `byteValue()` is the largest of those. I think it makes sense to change this condition to `==` and potentially also that of `GEOMETRY` while you're at it.

File: community/kernel/src/main/java/org/neo4j/kernel/impl/store/TemporalType.java [hunk 3]
```diff
@@ -77,9 +89,22 @@ public int calculateNumberOfBlocksUsedForTemporal( long firstBlock )
...
+                        LocalDate[] dates = new LocalDate[numbers.length()];
+                        for ( int i = 0; i < dates.length; i++ )
+                        {
+                            dates[i] = LocalDate.ofEpochDay( numbers.longValue( i ) );
+                        }
+                        return Values.dateArray( dates );
+                    }
+                    else
+                    {
+                        throw new RuntimeException( "Array with unexpected type. Actual:" + dataValue.getClass().getSimpleName() + ". Expected: LongArray." );
```
Comment (tinwelint) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/store/TemporalType.java]: Would be better to throw some more specific and coherent exception, like `InvalidRecordException` or the likes. Likewise for the other similar places.

**Review by tinwelint — CHANGES_REQUESTED** [review]
I think it looks good. Some minor findings, that's all

File: community/kernel/src/main/java/org/neo4j/kernel/impl/store/DynamicArrayStore.java (hunk 1 above)
Comment (sherfert) [reply community/kernel/src/main/java/org/neo4j/kernel/impl/store/DynamicArrayStore.java]: Indeed

**Review by tinwelint — APPROVED** [review]

**Pull request merged**